started garage

YonatanReisinger opened the pull request.

YonatanReisinger committed:
Started garage

YonatanReisinger committed:
made the adding of a new Vehicle to the Garage

YonatanReisinger committed:
made changes to factory and Vehicle.

added model name to Vehicle ctor.

michalmos262 commented on Ex03.GarageLogic/VehicleRecordInGarage.cs:
```diff
@@ -0,0 +1,57 @@
+                }
+                else
+                {
+                    throw new ArgumentException("Enum does not exits");
```
היית רושמת משהו ברור יותר:
Vehicle status is not correct

michalmos262 commented on Ex03.GarageLogic/VehicleRecordInGarage.cs:
```diff
@@ -0,0 +1,57 @@
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+namespace Ex03.GarageLogic
```
חסר בין לבין רווח

michalmos262 commented on Ex03.GarageLogic/Vehicle.cs:
```diff
@@ -52,37 +53,50 @@ public EnergyTank Engine
 
-        private bool areAllTiresAtCorrectMaxPressure(List<Tire> i_Tires)
+        public abstract int MaxTireAirPressure { get; }
+        public abstract int NumOfTires { get; }
```
חסר רווח בין לבין

michalmos262 commented on Ex03.GarageLogic/Vehicle.cs:
```diff
@@ -52,37 +53,50 @@ public EnergyTank Engine
-                m_Tires = i_Tires;
+                if (hasRequiredNumberOfTires(value) is false)
+                {
+                    //throw exception
```
הייתי שמה את ההערות האלה ב TODO זה יותר נוח לעקוב ככה אחרי מה צריך לעשות בויזואל

michalmos262 commented on Ex03.GarageLogic/VehicleFactory.cs:
```diff
@@ -12,24 +12,57 @@ public enum eVehicleType
         }
 
-        public Vehicle makeNewVehicle(eVehicleType i_VehicleType, string i_LisenceNumber)
+        public Vehicle MakeNewVehicle(eVehicleType i_VehicleType, string i_ModelName, string i_LisenceNumber)
```
למה היה צריך להוסיף פרמטר של model name?

YonatanReisinger replied:
כי זה אחד מהדברים שחייבים להיות ברכב מהתחלה

michalmos262 replied:
לפי מה שקראתי בהנחיות של התפריט למשתמש, רק המספר רישיון חובה כי כדי להכניס רכב למוסך המשתמש צריך להזין קודם מספר רישיון ואז המערכת בודקת האם הרכב הזה קיים,  ואם לא אז מבקשים פרטים כמו דגם, צבע אוטו וכו'...

YonatanReisinger replied:
אז כאילו נכניס רכב למוסך, ואחרי זה נכניס לו את המודל שלו? הראלי שאת שניהם אפשר להכניס ביחד לפני כל השאר שמכניסים. אלה אם כן לא הבנתי בדיוק את ההנחיות

michalmos262 replied:
תקרא את סעיף 1 בהנחיות של התפריט

michalmos262 commented on Ex03.GarageLogic/Garage.cs:
```diff
@@ -0,0 +1,118 @@
+
+        private void addVehicleToGarage(Vehicle i_Vehicle)
+        {
+            makeNewTiresForVehicle(i_Vehicle, )
```
חסר פה פרמטר

YonatanReisinger replied:
ידוע, בcommit הבא זה יהיה מתוקן

michalmos262 commented on Ex03.GarageLogic/Garage.cs:
```diff
@@ -0,0 +1,118 @@
+{
+    internal class Garage
+    {
+        private List<Vehicle> m_Vehicles;
+        private List<VehicleRecordInGarage> m_RecordsBook;
```
לא יותר יפה לעשות אולי מילון? שהמפתח זה מספר רישיון רכב והערך זה הנתונים עליו
לדעתי יותר קריא וגם פחות תחזוקה

YonatanReisinger replied:
האמת שחשבתי על זה, תסתכלי מה כתבתי בתגובה הבאה למה חשבתי לעשות את זה לא כמילון, אבל אני פתוח לשינויים אם זה נראים פחות הגיוני

michalmos262 replied:
ניראלי שעדיין מילון זה בול :)

michalmos262 commented on Ex03.GarageLogic/Garage.cs:
```diff
@@ -0,0 +1,118 @@
+        private List<Vehicle> m_Vehicles;
+        private List<VehicleRecordInGarage> m_RecordsBook;
+
+        public void AddVehicleToSystem(Vehicle i_Vehicle, string i_OwnerName, string i_OwnerPhoneNumber)
```
אולי לקרוא לפונקציה `AddVehicleToGarage` ?

YonatanReisinger replied:
חשבתי שיהיה כזאת של להכניס למערכת וכזאת שרק לגארז עצמו. כי רכב יכול להיות במערכת וכבר מתוקן ואז להכניס אותו לגארז אומר להכניס אותו לתיקון שוב.
כמו מוסך שיש לו רישום של כל מי שהיה אצלו אי פעם כדי לדעת על תקלות לאורך הזמן. כלומר לא תמיד ברכבים בגארז יהיה אותו מספר של איברים כמו במערך הרישומים.
עכשיו שאני חושב על זה זאת הסיבה שלא רציתי מילון חחחחחחחחחח

michalmos262 replied:
אני חושבת שזה סתם מסובך וזה מוסיף לוגיקה מיותרת.. לפי מה שכתוב הוספת רכב זה פשוט הוספה למוסך

michalmos262 commented on Ex03.GarageLogic/Garage.cs:
```diff
@@ -0,0 +1,118 @@
+                addVehicleToGarage(i_Vehicle);
+                addNewVehicleRecordToBook(i_Vehicle.LisenceNumber, i_OwnerName, i_OwnerPhoneNumber);
+            }
+            else
+            {
+                //exeption
+            }
```
לא חושבת שצריך פה exception

YonatanReisinger committed:
Changed garage to hold a dict instead of 2 arrays

YonatanReisinger committed:
the getVehicleByLisenceNumber throws an exception in case a non existing car is requested. thus i removed all the checks of vehicle == null

michalmos262 commented on Ex03.GarageLogic/Vehicle.cs:
```diff
@@ -52,37 +52,52 @@ public EnergyTank Engine
+
+        public abstract float MaxTimeBatteryCanLastInHours { get; }
+
+        public abstract FuelTank.eFuelType FuelType{ get; }
+
+
+
+        public override int GetHashCode()
```
רווחים מיותרים

michalmos262 commented on Ex03.GarageLogic/Garage.cs:
```diff
@@ -0,0 +1,92 @@
+            }
+            else
+            {
+                throw new Exception($"Vehicle with lisence number {i_LisenceNumber} does not exists");
```
exist *

michalmos262 commented on Ex03.GarageLogic/Garage.cs:
```diff
@@ -0,0 +1,92 @@
+            }
+            catch (Exception e)
+            {
+                throw new Exception("Can't make new tires for car that does not exists in the garage");
```
exist *

YonatanReisinger committed:
Added that every Vehicle class will know what specific details it needs and will know to verify them and set them as its values.

like that the UI and the Garage will be able to "talk" with the vehicles in a generic manner

YonatanReisinger committed:
Added following Garage methods:

2. get license numbers as a list according to status.
3. changing vehicle status
5. add fuel to vehicles
6. add electricity to vehicles

michalmos262 commented on Ex03.GarageLogic/Battery.cs:
```diff
@@ -7,9 +7,13 @@ public Battery(float i_MaximumTimeBatteryCanLast) : base(i_MaximumTimeBatteryCan
         }
 
-        public void Recharge(float i_AdditionalBatteryTimeInHours)
+        public override void ReEnegrize(float i_AdditionalEnergeyAmount, FuelTank.eFuelType? i_FuelType = null)
```
הכוונה ב ReEnergize?
